src/components/calculator.js

```diff
@@ -1,42 +1,60 @@
 import React from 'react';
-import '../App.css';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: '',
+      next: '',
+      operation: '',
+    };
+    this.onClickHandler = this.onClickHandler.bind(this);
   }
 
-  render() {
-    return (
-      <div className="calculator">
-        <div className="calculator-display">
-          0
-        </div>
-        <div className="calculator-keyboard">
-          <button type="button" className="calculator-key">AC</button>
-          <button type="button" className="calculator-key">+/-</button>
-          <button type="button" className="calculator-key">%</button>
-          <button type="button" className="calculator-key calculator-key-orange">÷</button>
-
-          <button type="button" className="calculator-key">7</button>
-          <button type="button" className="calculator-key">8</button>
-          <button type="button" className="calculator-key">9</button>
-          <button type="button" className="calculator-key calculator-key-orange">×</button>
+  onClickHandler(newValue) {
+    if (newValue.total != null) {
+      this.setState({ total: newValue.total });
+    }
 
-          <button type="button" className="calculator-key">4</button>
-          <button type="button" className="calculator-key">5</button>
-          <button type="button" className="calculator-key">6</button>
-          <button type="button" className="calculator-key calculator-key-orange">-</button>
+    this.setState({ next: newValue.next });
 
-          <button type="button" className="calculator-key">1</button>
-          <button type="button" className="calculator-key">2</button>
-          <button type="button" className="calculator-key">3</button>
-          <button type="button" className="calculator-key calculator-key-orange">+</button>
+    if (newValue.operation != null) {
+      this.setState({ operation: newValue.operation });
+    }
+  }
 
-          <button type="button" className="calculator-key calculator-key-zero">0</button>
-          <button type="button" className="calculator-key">.</button>
-          <button type="button" className="calculator-key calculator-key-orange">=</button>
+  render() {
+    const { total, next, operation } = this.state;
+    return (
+      <div className="calculator-container">
+        <div className="display-area">
+          <p className="display-results">
+            { total }
+            { operation }
+            { next }
```

Comment:
- [ ] OPTIONAL: Can you please make the default text in the display-results element to be '0'? The original design has the default display to '0'. It gives a better user experience. :pray: 
![image](https://user-images.githubusercontent.com/64894927/130145255-3955ef84-dfe4-4243-9002-86446a7428be.png)